essential-documentation/contribute-to-appflowy/software-contributions/environment-setup/building-on-macos.md

```diff
@@ -30,16 +31,22 @@ Skip this if flutter is already installed on your system.
 ./frontend/scripts/install_dev_env/install_macos.sh
 ```
 
-> FYI, AppFlowy uses [https://github.com/sagiegurari/cargo-make](https://github.com/sagiegurari/cargo-make) to construct the build scripts
+> FYI, AppFlowy uses [https://github.com/sagiegurari/cargo-make](https://github.com/sagiegurari/cargo-make) to construct the build scripts.
+> It is important that you add (dart) `pub` to $PATH, otherwise VS Code may error out. Add the following to your `.bashrc` or `.zshrc` in `$HOME`:
+> ```
+> export PATH="$PATH":"$HOME/.pub-cache/bin"
+> ```
+> Make sure to restart your terminal and VS Code
 
 ## **Step 4: Edit and run the application**
 
-1. Open the `frontend` folder located at xx/AppFlowy/frontend with VS Code.
-2. Go to the Run and Debug tab and then click AF: Clean + Rebuild All for the first time running.
+1. Open the `frontend` folder located at xx/AppFlowy/frontend with VS Code. It is important *not* to open the root folder, as that will not give access to the appropriate debug commands.
+2. Check the device selection, as of now AppFlowy only supports Desktop: 
+![device](https://user-images.githubusercontent.com/86001920/144546864-cebbf0c0-4eef-424e-93c7-e1e6b3a59669.png)
```

Comment:
Please upload your image to `../../../../.gitbook/assets/`

Reply:
I believe it is already there @vincentdchan I simply moved around an image that was already in the documentation. Am I wrong? 😄

Reply:
True. It seems that it's a legacy problem.